library/src/modules/backup/services/database-services/mysql.service.ts

```diff
@@ -0,0 +1,33 @@
+import { exec } from "child_process";
+import * as _os from "os";
+import { DatabaseConfig } from "../../models/shaman-backup.config";
+import { IDatabaseService } from "./backup-service.interface";
+import { injectable } from "inversify";
+
+@injectable()
+export class MysqlBackupService implements IDatabaseService {
+
+  type: string = 'mysql';
+
+  getBackup = (dbConfig: DatabaseConfig): Promise<string> => {
+    if (!this.validConfig(dbConfig))
+      return Promise.reject(new Error('Invalid config for mysql backup service. Database name, username and password are required.'));
+    return new Promise((resolve, reject) => {
+      let tmpDir = _os.tmpdir();
+      let command = `mysqldump -u ${dbConfig.username} -p${dbConfig.password} ${dbConfig.name} > ${tmpDir}/${dbConfig.name}.sql`;
+      exec(command, (error, _stdout, _stderr) => {
+        if (error)
+          reject(new Error(error.message));
```

Comment:
you will want to return here as well, so it doesn't then call 'resolve' immediately afterwards. For example:

```ts
return reject(new Error(error.message));
```